enterprise/utils.py

```diff
@@ -1091,3 +1093,92 @@ def delete_data_sharing_consent(course_id, customer_uuid, user_email):
     # Deleting the DCS cache
     consent_cache_key = get_cache_key(type='data_sharing_consent_needed', user_id=user.id, course_id=course_id)
     TieredCache.delete_all_tiers(consent_cache_key)
+
+
+def create_tableau_user(username, enterprise_customer_user):
+    """
+    Create the user in tableau and store the tableau user if in the EnterpriseAnalyticsUser model.
+    """
+    from enterprise.models import EnterpriseAnalyticsUser  # pylint: disable=import-outside-toplevel
+    tableau_auth = TSC.TableauAuth(settings.TABLEAU_ADMIN_USER, settings.TABLEAU_ADMIN_USER_PASSWORD)
+    server = TSC.Server(settings.TABLEAU_URL)
+    enterprise_uuid = enterprise_customer_user.enterprise_customer.uuid
+    try:
+        with server.auth.sign_in(tableau_auth):
+            user_item = TSC.UserItem(username, TSC.UserItem.Roles.Viewer)
+            user = server.users.add(user_item)
+            enterprise_analytics_user = EnterpriseAnalyticsUser.objects.create(
+                enterprise_customer_user=enterprise_customer_user,
+                analytics_user_id=user.id
+            )
+            enterprise_analytics_user.save()
+            add_user_to_tableau_group(user.id, str(enterprise_uuid))
+    except ServerResponseError as exc:
+        LOGGER.info(
+            '[TABLEAU USER SYNC] Could not sync enterprise admin user %s in tableau.'
+            'Server returned: %s', username, str(exc),
+        )
+
+
+def add_user_to_tableau_group(tableau_user_id, tableau_group):
+    """
+    Associate a user with the group in tableau. Create the group if it does not already exist.
+    """
+    tableau_auth = TSC.TableauAuth(settings.TABLEAU_ADMIN_USER, settings.TABLEAU_ADMIN_USER_PASSWORD)
+    server = TSC.Server(settings.TABLEAU_URL)
+    try:
+        with server.auth.sign_in(tableau_auth):
+            group_to_create = TSC.GroupItem(tableau_group)
+            group = server.groups.create(group_to_create)
+            server.groups.add_user(group, tableau_user_id)
+    except ServerResponseError as exc:
+        if exc.code == '409009':  # Group already exists
+            try:
+                with server.auth.sign_in(tableau_auth):
+                    enterprise_group = TSC.GroupItem(tableau_group)
+                    server.groups.add_user(enterprise_group, tableau_user_id)
+            except ServerResponseError as exc:
+                LOGGER.info(
+                    '[TABLEAU USER SYNC] Could not associate user %s with group %s in tableau.'
+                    'Server returned: %s', tableau_user_id, tableau_group, str(exc),
+                )
+        else:
+            raise
+
+
+def delete_tableau_user(enterprise_customer_user):
```

Comment:
Can we refactor this to call `delete_tableau_user_by_id()` to reduce code duplication.

And, since we are also using EnterpriseAnalyticsUser in `create_tableau_user()', we should just move the import to the top of the file.

Finally, we should be sure to log a message in this method if the 'enterprise_analytics_user' is None.

Reply:
local import was required to fix a circular import